tests/command_line/test_cosym.py

```diff
@@ -49,37 +49,57 @@ def test_cosym(dials_data, run_in_tmpdir, space_group, engine):
 
 def test_cosym_partial_dataset(dials_data, run_in_tmpdir):
     """Test how cosym handles partial/bad datasets."""
-    mcp = dials_data("multi_crystal_proteinase_k")
+    mcp = dials_data("multi_crystal_proteinase_k", pathlib=True)
     args = []
     for i in [1, 2]:
-        args.append(mcp.join("experiments_%d.json" % i).strpath)
-        args.append(mcp.join("reflections_%d.pickle" % i).strpath)
+        args.append(str(mcp / f"experiments_{i}.json"))
+        args.append(str(mcp / f"reflections_{i}.pickle"))
```

Comment:
I think `os.fspath` is generally recommended vs `str` in cases where this is necessary:
https://docs.python.org/3/library/os.html#os.fspath
Is it actually necessary here? If so, we should probably make the underlying dials/dxtbx file-handling code support `PathLike` objects (but outside the scope of this PR).
```suggestion
        args.append(os.fspath(mcp / f"experiments_{i}.json"))
        args.append(os.fspath(mcp / f"reflections_{i}.pickle"))
```

Reply:
In all these cases, it's obvious from reading that it's a Path object  or some kind of smart-path. While `os.fspath` is technically correct, it does make it look strange. My brain: "str - that's nice and obvious". "fspath - why is this necessary instead of str"